examples/ex18.hpp

```diff
@@ -0,0 +1,558 @@
+//                  MFEM Example 18 - Serial/Parallel Shared Code
```

Comment:
Style question. There is a lot of actual implementation in this .hpp file. I understand that it is really the shared serial and parallel stuff, but should there be a separate .cpp file that has the actual function implementations? I don't really have a good reason, it's just the way I'm used to seeing things. What do you think @tzanio ?

Reply:
We usually just duplicate the code that is shared between the serial and parallel versions of the example, but in this case I agree that having it in a separate file also makes sense. If this was a miniapp, I'd say split `ex18.hpp` into `ex18_common.hpp` and `ex18_common.cpp` and compile and link with the latter. For the examples though, I'd prefer if each is just a single compilation, and then if we are `#include`-ing it really doesn't matter what the file extension is. So `ex18.hpp` works for me. It is not like we don't have implementations inside header files already :).